server/src/main/java/com/netflix/conductor/server/ConductorServer.java

```diff
@@ -267,6 +278,43 @@ public synchronized void start(int port, boolean join) throws Exception {
 
     }
 
+    private static void shutdownListener(Injector iocInjector) {
```

Comment:
can we moved the Signal handling and the DAO shutdown to the ServerShutdown class

Q: if a process takes a long time to terminate, will it stall the shutdown or is there a time frame after which the container will be forcibly removed?

Reply:
Thanks 👍 
Lets talk tomorrow on the first point.